modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -801,46 +849,56 @@ class StreamingOutput final: public cv::gimpl::GIslandExecutable::IOutput
         GAPI_ITT_STATIC_LOCAL_HANDLE(outputs_post_hndl, "StreamingOutput::post");
         GAPI_ITT_AUTO_TRACE_GUARD(outputs_post_hndl);
 
-        std::lock_guard<std::mutex> lock{m_mutex};
-
-        // Mark the output ready for posting. If it is the first in the line,
-        // actually post it and all its successors which are ready for posting too.
-        auto it = m_postIdx.find(cv::gimpl::proto::ptr(argp));
-        GAPI_Assert(it != m_postIdx.end());
-        const int out_idx = it->second.first;
-        const auto out_iter = it->second.second;
-        out_iter->ready = true;
-        m_postIdx.erase(it); // Drop the link from the cache anyway
-        if (out_iter != m_postings[out_idx].begin())
+        bool stop_triggered = false;
```

Comment:
It should be under `mutex` as well, right?

Reply:
So, we only need this flag for logging?

Reply:
it should be not under mutex protection because it is not shared resource but stack-variable

Reply:
Yes, we do: Stop event is rare in comparison with regular event and require special processing - thus it is worth to know about it.
This slight refactoring is used to make printout GAPI_LOG_INFO without mutex protection  to prevent pessimization